AstSemantics.md

```diff
@@ -231,23 +228,22 @@ others, etc.
 ## Control flow structures
 
 WebAssembly offers basic structured control flow with the following constructs.
-All control flow structures, except `case`, are statements.
-
- * `block`: a fixed-length sequence of statements with a label at the end
- * `loop`: a fixed-length sequence of statements with a label at the end
-           and a loop header label at the top (note: this does not loop by
-           itself, so one would often combine this with a br_if at the end
-           to form a branch back to the top)
- * `if`: if statement with then body
- * `if_else`: if statement with then and else bodies
- * `br`: branch to a given label in an enclosing construct (see below)
+Since all AST nodes are expressions in WebAssembly, control constructs may yield
+a value and appear as children of other expressions.
+
+ * `block`: a fixed-length sequence of expressions with a label at the end
+ * `loop`: a block with an additional label at the beginning which may be used to form loops
```

Comment:
I also think it's worth leaving the note in there about loops needing an explicit branch to the top.